### ISSUE TITLE
Create compat

### ISSUE THREAD

**Issue by Yarden-zamir** [jaquadro/StorageDrawers#969]
Because of the way this mod stores information and the way create reads it, there are a few incompatibilities between the two

- stockpile switch
> - doesn't work on 1x1 drawer. 
> - works on 2x2 but detects only up to two stacks. 
> - works on 4x4 but detects only up to 4 stacks
- contraptions
> - don't detect drawers as inventories for storage

**Comment by Yarden-zamir** [comment]
https://github.com/Creators-of-Create/Create/issues/2239 related issiue in create github

**Comment by jaquadro** [comment]
For stockpile switch, it looks like they artificially limit their handling to a normal minecraft stack size.  It's valid for IItemHandler imeplementations to report stack sizes much larger than that, and SD does as it's a kind of deep storage mod.

Not sure about the contraptions, but if they're trying to interface with Minecraft-style, this mod is not compatible with that interface.  It's only accessible through Forge's IItemHandler.

**Comment by emifloofy** [comment]
> Not sure about the contraptions, but if they're trying to interface with Minecraft-style, this mod is not compatible with that interface. It's only accessible through Forge's IItemHandler.

Seems like the mod provides its own API for storage on contraptions according to https://github.com/Creators-of-Create/Create/issues/7986

I haven't been able to find any documentation for the API itself, but apparently there's `#create:chest_mounted_storage` and `#create:simple_mounted_storage` for simple solutions. I doubt that those apply here though, as upgrades could cause potential problems.

After digging through the code for a bit I was able to find [this file](https://github.com/Creators-of-Create/Create/blob/mc1.20.1/dev/src/main/java/com/simibubi/create/api/contraption/storage/item/MountedItemStorage.java) which might be of help, but I have zero experience writing mods, so I have no idea.

**Comment by electricsteve** [comment]
> > Not sure about the contraptions, but if they're trying to interface with Minecraft-style, this mod is not compatible with that interface. It's only accessible through Forge's IItemHandler.
> 
> Seems like the mod provides its own API for storage on contraptions according to [Creators-of-Create/Create#7986](https://github.com/Creators-of-Create/Create/issues/7986)
> 
> I haven't been able to find any documentation for the API itself, but apparently there's `#create:chest_mounted_storage` and `#create:simple_mounted_storage` for simple solutions. I doubt that those apply here though, as upgrades could cause potential problems.
> 
> After digging through the code for a bit I was able to find [this file](https://github.com/Creators-of-Create/Create/blob/mc1.20.1/dev/src/main/java/com/simibubi/create/api/contraption/storage/item/MountedItemStorage.java) which might be of help, but I have zero experience writing mods, so I have no idea.

Yes it now has a simple to use api. For example P3pp3rF1y already did it with their sophisticated mods, which also has drawers. [Github Repo](https://github.com/P3pp3rF1y/SophisticatedStorageCreateIntegration). They seem to have taken not that much time to do it, which makes me think create integration can be done in a weekend by myself (I know some modding, but am by no means an expert), but I also want to do other things, and I think @jaquadro can do it way faster (in terms of coding speed). But if they can not or will not do it soon I may do it. (if they don't react within the next week I may just do it because I have vacation)

**Comment by electricsteve** [comment]
I tried today with block tags, but I can't for the life of me get the tag to load, I'm probably doing something wrong but I can't figure it out for the life of me.
